includes/object-cache.php

```diff
@@ -636,6 +640,116 @@ protected function connectUsingPredis( $parameters ) {
         );
     }
 
+    /**
+     * Connect to Redis using the Credis library.
+     *
+     * @param  array $parameters
+     * @return void
+     */
+    protected function connectUsingCredis( $parameters ) {
+        $client = 'Credis';
+
+        // Require PHP 5.4 or greater
+        if ( version_compare( PHP_VERSION, '5.4.0', '<' ) ) {
+            throw new Exception( 'Predis requires PHP 5.4 or newer.' );
+        }
+
+        $creds_path = sprintf(
+            '%s/redis-cache/dependencies/vendor/colinmollenhour/credis/',
+            defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins'
+        );
+
+        $to_load = [];
+
+        // Load bundled Credis library.
+        if ( ! class_exists( 'Credis_Client' ) ) {
+            $to_load[] = 'Client.php';
+        }
+
+        if ( ( defined( 'WP_REDIS_SHARDS' ) || defined( 'WP_REDIS_SENTINEL' ) || defined( 'WP_REDIS_SERVERS' ) || defined( 'WP_REDIS_CLUSTER' ) ) && ! class_exists( 'Credis_Cluster' ) ) {
+            $to_load[] = 'Cluster.php';
+
+            if ( true
+                && defined( 'WP_REDIS_SENTINEL' )
+                && ! class_exists( 'Credis_Sentinel' )
+            ) {
+                $to_load[] = 'Sentinel.php';
+            }
+        }
+
+        foreach ( $to_load as $sub_path ) {
+            $path = $creds_path . $sub_path;
+            if ( file_exists( $path ) ) {
+                require_once $path;
+            } else {
+                throw new Exception(
+                    sprintf(
+                        'Credis "%s" library not found. Re-install Redis Cache plugin or delete object-cache.php.',
+                        str_replace( '.php', '', $sub_path )
+                    )
+                );
+            }
+        }
+
+        if ( defined( 'WP_REDIS_SHARDS' ) ) {
+            // TODO: untested!
+            // Sharding not supported?
```

Comment:
Maybe we should throw for all unsupported configurations (PhpRedis needs that too). Thoughts?

Reply:
A configuration checker class would be ideal. I almost started writing one a couple of days ago :wink:.
If we allow configuration via UI this would be necessary.